src/main/java/net/dries007/tfc/objects/te/TEPowderKeg.java

```diff
@@ -0,0 +1,248 @@
+ */
+
+package net.dries007.tfc.objects.te;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import javax.annotation.ParametersAreNonnullByDefault;
+
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.entity.EntityLivingBase;
+import net.minecraft.init.Blocks;
+import net.minecraft.init.Items;
+import net.minecraft.init.SoundEvents;
+import net.minecraft.inventory.InventoryHelper;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.SoundCategory;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import net.minecraftforge.common.capabilities.Capability;
+
+import net.minecraftforge.items.CapabilityItemHandler;
+import net.minecraftforge.items.ItemStackHandler;
+
+import net.dries007.tfc.Constants;
+import net.dries007.tfc.objects.blocks.BlockPowderKeg;
+import net.dries007.tfc.objects.inventory.capability.IItemHandlerSidedCallback;
+import net.dries007.tfc.objects.inventory.capability.ItemHandlerSidedWrapper;
+
+import static net.dries007.tfc.objects.blocks.BlockPowderKeg.SEALED;
+
+/**
+ * @see BlockPowderKeg
+ */
+@ParametersAreNonnullByDefault
+public class TEPowderKeg extends TETickableInventory implements IItemHandlerSidedCallback
+{
+    private boolean sealed;
+    private int fuse = -1;
+
+    private boolean isLit = false;
+    private EntityLivingBase igniter;
+
+    public TEPowderKeg()
+    {
+        super(new ItemStackHandler(12));
+    }
+
+    /**
+     * Called when this TileEntity was created by placing a sealed keg Item.
+     * Loads its data from the Item's NBTTagCompound without loading xyz coordinates.
+     *
+     * @param nbt The NBTTagCompound to load from.
+     */
+    public void readFromItemTag(NBTTagCompound nbt)
+    {
+        inventory.deserializeNBT(nbt.getCompoundTag("inventory"));
+        sealed = nbt.getBoolean("sealed");
+        markForSync();
+    }
+
+    /**
+     * Called once per side when the TileEntity has finished loading.
+     * On servers, this is the earliest point in time to safely access the TE's World object.
+     */
+    @Override
+    public void onLoad()
+    {
+        if (!world.isRemote)
+        {
+            sealed = world.getBlockState(pos).getValue(SEALED);
+        }
+    }
+
+    @Override
+    public boolean canInsert(int slot, ItemStack stack, EnumFacing side)
+    {
+        return !world.getBlockState(pos).getValue(SEALED) && isItemValid(slot, stack);
+    }
+
+    @Override
+    public boolean canExtract(int slot, EnumFacing side)
+    {
+        return !sealed;
+    }
+
+    public void onSealed()
+    {
+        // Update sealed tick info and sync to client
+        sealed = true;
+        markForSync();
+    }
+
+    public void onUnseal()
+    {
+        // Update sealed info and sync to client
+        sealed = false;
+        markForSync();
+    }
```

Comment:
```java
public void setSealed(boolean)
{
    this.sealed = sealed;
    markForSync();
}
```
makes this a damsight simpler, no?

Reply:
Consistency. Barrel and Large Vessel are implemented this way, with additional logic. Do we keep the same interface for sealable inventories, or accept this simplification for powderkegs?

Reply:
> [Other unrelated blocks] are implemented this way, with additional logic

And that's why, because they do separate seal + unseal checks. This does not. There is no public API or interface that must be adhered to. Consistency does not mean copy paste - as I've pointed out in a number of other places, code that is the same does not mean it's the best possible solution to two *different problems*. :)

Reply:
Thanks for feedback. Code updated. Apologize for the clumsiness.